backstory/src/test/java/com/google/sps/images/AnnotatedImageTest.java

```diff
@@ -0,0 +1,159 @@
+   *
+   * @param reference the local reference of the image.
+   * @param fileType the file extension of the image.
+   * @return the byte array representation of the image.
+   */
+  private static byte[] getBytesFromImageReference(String reference, String fileType) throws IOException {
+    ByteArrayOutputStream bos = new ByteArrayOutputStream();
+
+    try {
+      BufferedImage bImage = ImageIO.read(new File(reference));
+      ImageIO.write(bImage, fileType, bos);
+    } catch (IOException exception){
+      throw exception;
+    }
+
+    return bos.toByteArray();
+  }
+
+  /**
+   * public AnnotatedImage(byte[] rawImageData, List<EntityAnnotation> labelAnnotations);
+   *  rawImageData is null.
+   *  labelAnnotations is empty.
+   *  Expects IllegalArgumentException to be thrown.
+  */
+  @Test(expected = IllegalArgumentException.class)
+  public void constructorNullImageData() throws IllegalArgumentException {
+    AnnotatedImage annotatedImageActual = new AnnotatedImage(nullRawImageData, emptyLabelAnnotations);
+  }
+
+  /**
+   * public AnnotatedImage(byte[] rawImageData, List<EntityAnnotation> labelAnnotations);
+   *  rawImageData is empty.
+   *  labelAnnotations is non-empty.
+   *  Expects IllegalArgumentException to be thrown.
+  */
+  @Test(expected = IllegalArgumentException.class)
+  public void constructorEmptyImageData() throws IllegalArgumentException {
+    AnnotatedImage annotatedImageActual = new AnnotatedImage(emptyRawImageData, labelAnnotations);
+  }
+
+  /**
+   * public AnnotatedImage(byte[] rawImageData, List<EntityAnnotation> labelAnnotations);
+   *  rawImageData is non-empty image data.
+   *  labelAnnotations is non-empty.
+  */
+  @Test
+  public void constructorRawImageData() throws IllegalArgumentException {
+    AnnotatedImage actual = new AnnotatedImage(rawImageData, labelAnnotations);
+    assertTrue(Arrays.equals(rawImageData, actual.getRawImageData()));
+    assertEquals(labelAnnotations, actual.getLabelAnnotations());
+  }
+
+  /**
+   * Tests paritions on getLabelDescriptions();
+   *  labelAnnotations is empty.
+  */
+  @Test
+  public void getLabelDescriptionsEmptyLabelAnnotations() throws IllegalArgumentException {
+    AnnotatedImage annotatedImageActual = new AnnotatedImage(rawImageData, emptyLabelAnnotations);
+    List<String> actual = annotatedImageActual.getLabelDescriptions();
+    List<String> expected = new ArrayList<>();
+    assertEquals(expected, actual);
+  }
+
+  /**
+   * Tests paritions on getLabelDescriptions();
+   *  labelAnnotations is a non-empty list of EntityAnnotations.
+  */
+  @Test
+  public void getLabelDescriptionsNonEmptyLabelAnnotations() throws IllegalArgumentException {
+    AnnotatedImage annotatedImageActual = new AnnotatedImage(rawImageData, labelAnnotations);
+    List<String> actual = annotatedImageActual.getLabelDescriptions();
+    List<String> expected = new ArrayList<>();
+    expected.add("DescriptionOne");
+    expected.add("DescriptionOne");
```

Comment:
When reading this test, I wonder where "DescriptionOne" comes from. Ideally readers don't have to navigate outside of a test case to understand a test case. Can that be set during this test?

```
String description = "DescriptionOne";
// This would be a private helper
EntityAnnotation entityAnnotation = makeEntityAnnotationWithDescription(description);
List<EntityAnnotation> annotations = ...;
...;
assertEquals(description, annotatedImage.getLabelDescriptions.get(0));
```

This would allow me to read through this without searching for "DescriptionOne".

Reply:
Sure, I will make a private method to create MockEntityAnnotationWithDescription(description).